packages/rust_gtk_pill/test.sh

```
@@ -1,5 +1,337 @@
+  echo '--- Assistant: using tool ---' >&2
+  echo '{"type":"assistant_state","active":true,"input_mode":"voice","compact":false,"conversation_id":"conv_1","user_prompt":"Tell me about the weather ... in San Francisco today please","messages":[{"id":"m1","content":null,"is_error":false,"is_tool_result":false,"tool_name":null,"tool_description":null,"reason":null}],"streaming":{"message_id":"m1","tool_calls":[{"id":"tc1","name":"get_weather","done":false}],"reasoning":"Let me look up the current weather...","is_streaming":true},"permissions":[]}'
+  sleep 1.5
+
+  # Response
+  echo '--- Assistant: response ---' >&2
+  echo '{"type":"assistant_state","active":true,"input_mode":"voice","compact":false,"conversation_id":"conv_1","user_prompt":"Tell me about the weather ... in San Francisco today please","messages":[{"id":"m1","content":"It'\''s currently **62°F** (17°C) in San Francisco with partly cloudy skies.","is_error":false,"is_tool_result":false,"tool_name":null,"tool_description":null,"reason":null}],"streaming":null,"permissions":[]}'
+  sleep 2
+
+  # Multiple messages
+  echo '--- Assistant: multiple messages ---' >&2
+  echo '{"type":"assistant_state","active":true,"input_mode":"voice","compact":false,"conversation_id":"conv_1","user_prompt":"Tell me about the weather ... in San Francisco today please","messages":[{"id":"m1","content":"It'\''s currently **62°F** (17°C) in San Francisco with partly cloudy skies.","is_error":false,"is_tool_result":false,"tool_name":null,"tool_description":null,"reason":null},{"id":"m2","content":null,"is_error":false,"is_tool_result":true,"tool_name":"get_forecast","tool_description":"Get weather forecast","reason":"Checking weekly outlook"},{"id":"m3","content":"The forecast for the rest of the week:\n\n- Tuesday: 65°F, sunny\n- Wednesday: 58°F, fog\n- Thursday: 61°F, partly cloudy\n- Friday: 63°F, clear","is_error":false,"is_tool_result":false,"tool_name":null,"tool_description":null,"reason":null}],"streaming":null,"permissions":[]}'
```

Shellcheck (0.11.0)

[info] echo may not expand escape sequences. Use printf.

(SC2028)